File: clients/js/src/features/config_interface_implementations.ts
import { Input } from "../input";

import { PolywrapClientConfigBuilder, PolywrapClient } from "@polywrap/client-js";

export async function runTestCase(input: unknown): Promise<void> {
  const inputObj = Input.expectObject<{
    interfaceUri: unknown;
    implementations: unknown;
  }>(input);
  const interfaceUri = Input.expectUri(inputObj.interfaceUri).uri;
  const implementations = Input.expectArray<string>(inputObj.implementations);

  console.log("Adding Interface Implementations to ClientConfig");

  const config = new PolywrapClientConfigBuilder()
    .addInterfaceImplementations(
      interfaceUri,
      implementations
    )
    .build();

  const client = new PolywrapClient(config);

  console.log("Getting Implementations");

  const result = await client.getImplementations(interfaceUri);

  if (result.ok && result.value.length > 0) {
    console.log(`Found ${result.value.length} Implementations`);
    console.log("Success!");
  }
}

File: clients/py/src/features/config_interface_implementations.py
from typing import Any, List, cast
from polywrap_client import PolywrapClient
from polywrap_client_config_builder import PolywrapClientConfigBuilder
from polywrap_core import Uri

from pydantic import BaseModel, Field

from validators import UriStr


class TestCaseInput(BaseModel):
    interface: UriStr = Field(..., alias="interfaceUri")
    implementations: List[UriStr]


def run_test_case(input: Any) -> None:
    input_obj = TestCaseInput.validate(input)

    print("Adding Interface Implementations to ClientConfig")

    config = (
        PolywrapClientConfigBuilder()
        .add_interface_implementations(
            input_obj.interface, cast(List[Uri], input_obj.implementations)
        )
        .build()
    )
    client = PolywrapClient(config)

    print("Getting Implementations")

    try:
        result = client.get_implementations(input_obj.interface)
        if result and len(result) > 0:
            print(f"Found {len(result)} Implementations")
            print("Success!")
    except Exception as e:
        print(e)
